lib/service_actor/arguments_validator.rb

```diff
@@ -3,6 +3,14 @@
 module ServiceActor::ArgumentsValidator
   module_function
 
+  def validate_origin_name(name, origin:)
+    return unless ServiceActor::Result.instance_methods.include?(name.to_sym)
```

Comment:
In an app I have this brings up a lot of methods that _could_ make sense as arguments in the app and might not ever bring up issues:

```
as_json
clone
deep_dup
delete_field
dig
display
dup
each_pair
encode_with
enum_for
extend
freeze
gem
hash
itself
marshal_dump
presence
presence_in
table
taint
tap
then
to_enum
to_json
to_param
to_query
to_s
to_yaml
trust
try
untaint
untrust
with
with_options
```

I’m not sure how to handle this. I’d rather not outright disallow all of these :/

To me it would be nice to be able to do `OrderExport.call(order:, to_json: true, with: [:price, :date], delete_field: :id,  display: false, table: true, try: 2.seconds)` without having to find new names for those arguments. 

Perhaps your idea of using a `BasicObject` instead might be a good choice to allow more methods?

Reply:
That's a valid point. Note that after ostruct removal the actual blacklisted methods list is

`ServiceActor::Result.instance_methods - OpenStruct.instance_methods(false)`

> Perhaps your idea of using a BasicObject instead might be a good choice to allow more methods?

I'll look into this

Reply:
Just for the reference:

```
[19] pry(main)> BasicObject.instance_methods
[
    [0]             !()      
    [1]            !=(arg1)  
    [2]            ==(arg1)  
    [3]   __binding__()      
    [4]        __id__()      
    [5]      __send__(*arg1) 
    [6]        equal?(arg1)  
    [7] instance_eval(*arg1) 
    [8] instance_exec(*arg1) 
]
[20] pry(main)> RUBY_VERSION
"3.2.3"
```

Reply:
I like this idea, and love seeing ostruct removed (I remember from years ago that it is very slow).